src/libs/hovercard/hovercard.js

```diff
@@ -0,0 +1,184 @@
+import { fetchProfile, fetchRelationship } from "../fetchPronouns.js";
+import { addShowMoreButton, generateProfile } from "./hovercard_generators.js";
+import { normaliseAccountName } from "../protootshelpers.js";
+import { hoverCardSettings } from "../settings.js";
+import { createElementWithClass } from "./hovercard_helpers.js";
+import { runtime } from "webextension-polyfill";
+
+const listenerTimeout = 200;
+
+let layer;
+let hovering = null;
+let hovercardExists = false;
+
+let mousePos = { x: undefined, y: undefined };
+
+/**
+ * Adds hovercard layer div to document body
+ */
+export function addHoverCardLayer() {
+	if (document.querySelector("#hovercardlayer")) return;
+
+	layer = createElementWithClass("div", "protoots-hovercardlayer");
+	layer.id = "hovercardlayer";
+
+	document.body.appendChild(layer);
+
+	window.addEventListener("mousemove", (event) => {
+		mousePos = { x: event.clientX, y: event.clientY };
+	});
+}
+
+/**
+ * Adds mouse enter and leave Eventlisteners to given element.
+ * Eventlisteners wait 200ms before adding or removing a hoverCard
+ * @param {HTMLElement} el Element to add Eventlisteners to
+ * @returns
+ */
+export function addHoverCardListener(el) {
+	if (el.nodeName != "ARTICLE") return;
+
+	const nametag = /** @type {HTMLElement|null} */ (el.querySelector(".display-name"));
+	const id = el.dataset.id;
+
+	nametag.addEventListener("mouseenter", () => {
+		hovering = nametag;
+		const enterpos = { x: mousePos.x, y: mousePos.y };
+		setTimeout(() => {
+			// if the cursor has moved more than 50 pixels since
+			// entering the name, don't generate a card
+			if (Math.hypot(mousePos.x - enterpos.x, mousePos.y - enterpos.y) > 50) return;
+			addHoverCard(nametag, id);
+		}, listenerTimeout);
+	});
+
+	nametag.addEventListener("mouseleave", () => {
```

Comment:
not sure, did you try to enter the nametag a little bit and then move the cursor out of it on the same path?
can you get the hovercard stuck with that?
(hinting of race between mouseleave and the timeout-function in mouseenter; resolve via saving timeoutID and calling clearTimeout)

Reply:
knowing of the existance of clearTimeout helps a lot, thanks :D